transport/src/main/java/io/netty/channel/nio/AbstractNioByteChannel.java

```diff
@@ -126,6 +126,9 @@ public final void read() {
                         byteBuf.release();
                         byteBuf = null;
                         close = allocHandle.lastBytesRead() < 0;
+                        if (close) {
+                            readPending = false;
```

Comment:
@Scottmitch can you just please add a comment why we handle it special here compared to the native transports ?

Reply:
done